src/db/collections/block.rs

```diff
@@ -276,15 +345,15 @@ impl MongoDb {
     /// Create aggregate statistics of all addresses.
     pub async fn aggregate_addresses(
         &self,
-        start_milestone: MilestoneIndex,
-        end_milestone: MilestoneIndex,
+        start_milestone: crate::types::tangle::MilestoneIndex,
+        end_milestone: crate::types::tangle::MilestoneIndex,
     ) -> Result<Option<AddressAnalyticsResult>, Error> {
         Ok(self.0.collection::<BlockDocument>(BlockDocument::COLLECTION)
         .aggregate(
             vec![
                 doc! { "$match": {
                     "inclusion_state": LedgerInclusionState::Included,
-                    "milestone_index": { "$gt": start_milestone, "$lt": end_milestone },
+                    "metadata.referenced_by_milestone_index": { "$gt": start_milestone, "$lt": end_milestone },
```

Comment:
that is a non-inclusive range, right? Are we sure about that?

Reply:
Changed this while adding the paging state